pkg/query-service/app/preferences/model.go

```diff
@@ -0,0 +1,479 @@
+		if !ok {
+			return p.ErrorValueTypeMismatch()
+		}
+	}
+
+	// check the validity of the value being part of allowed values or the range specified if any
+	if p.IsDiscreteValues {
+		if p.AllowedValues != nil {
+			isInAllowedValues := p.checkIfInAllowedValues(preferenceValue)
+			if !isInAllowedValues {
+				return &model.ApiError{Typ: model.ErrorBadData, Err: fmt.Errorf("the preference value is not in the list of allowedValues: %v", p.AllowedValues)}
+			}
+		}
+	}
+	return nil
+}
+
+func (p *Preference) IsEnabledForScope(scope string) bool {
+	isPreferenceEnabledForGivenScope := false
+	if p.AllowedScopes != nil {
+		for _, allowedScope := range p.AllowedScopes {
+			if allowedScope == scope {
+				isPreferenceEnabledForGivenScope = true
+			}
+		}
+	}
+	return isPreferenceEnabledForGivenScope
+}
+
+func (p *Preference) SanitizeValue(preferenceValue interface{}) interface{} {
+	switch p.ValueType {
+	case PreferenceValueTypeBoolean:
+		if preferenceValue == "1" {
+			return true
+		} else {
+			return false
+		}
+	default:
+		return preferenceValue
+	}
+}
+
+type AllPreferences struct {
+	Preference
+	Value interface{} `json:"value"`
+}
+
+type PreferenceKV struct {
+	PreferenceId    string      `json:"preference_id" db:"preference_id"`
+	PreferenceValue interface{} `json:"preference_value" db:"preference_value"`
+}
+
+type UpdatePreference struct {
+	PreferenceValue interface{} `json:"preference_value"`
+}
+
+var db *sqlx.DB
+
+func InitDB(datasourceName string) error {
+	var err error
+	db, err = sqlx.Open("sqlite3", datasourceName)
+
+	if err != nil {
+		return err
+	}
+
+	// create the user preference table
+	tableSchema := `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS user_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		user_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,user_id),
+		FOREIGN KEY (user_id)
+			REFERENCES users(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(tableSchema)
+	if err != nil {
+		return fmt.Errorf("error in creating user_preference table: %s", err.Error())
+	}
+
+	// create the org preference table
+	tableSchema = `
+	PRAGMA foreign_keys = ON;
+	CREATE TABLE IF NOT EXISTS org_preference(
+		preference_id TEXT NOT NULL,
+		preference_value TEXT,
+		org_id TEXT NOT NULL,
+		PRIMARY KEY (preference_id,org_id),
+		FOREIGN KEY (org_id)
+			REFERENCES organizations(id)
+			ON UPDATE CASCADE
+			ON DELETE CASCADE
+	);`
+
+	_, err = db.Exec(tableSchema)
+	if err != nil {
+		return fmt.Errorf("error in creating org_preference table: %s", err.Error())
+	}
+
+	return nil
+}
+
+// org preference functions
+func GetOrgPreference(ctx context.Context, preferenceId string, orgId string) (*PreferenceKV, *model.ApiError) {
+	// check if the preference key exists or not
+	preference, seen := preferenceMap[preferenceId]
+	if !seen {
+		return nil, &model.ApiError{Typ: model.ErrorBadData, Err: fmt.Errorf("no such preferenceId exists: %s", preferenceId)}
```

Comment:
ErrorNotFound is more appropriate?

```suggestion
		return nil, &model.ApiError{Typ: model.ErrorNotFound, Err: fmt.Errorf("no such preferenceId exists: %s", preferenceId)}
```